src/en/news/blog/2024/ceph-a-journey-to-1tibps/index.md

```diff
@@ -0,0 +1,345 @@
+
+![](images/All_Fixes_Validation_Tests_-_FIO_4MB_Throughput.svg)
+![](images/All_Fixes_Validation_Tests_-_FIO_4KB_IOPS.svg)
+
+4KB random write performance was still lower than I wanted it to be, but at least now we were in roughly the right ballpark given that we had fewer OSDs, only 3/5 the number of nodes, and fewer (though faster) cores per OSD.  At this point we were nearing winter break.  The customer wanted to redeploy the OS to the correct boot drives and update the deployment with all of the fixes and tunings we had discovered.  The plan was to take off the holiday break, then spend the first week of the new year finishing the burn-in tests.  Hopefully we could start migrating the cluster the following week.
+
+# The First week of 2024 
+
+On the morning of January 2nd I logged into slack and was greeted by a scene I'll describe as moderately controlled chaos.  A completely different cluster we are involved in was having a major outage.  Without getting too into the details, it took 3 days pull that cluster back from the brink and get it into a stable and relatively healthy state.  It wasn't until Friday that I was able to get back to performance testing.  I was able to secure an extra day for testing on Monday, but this meant I was under a huge time crunch to showcase that the cluster could perform well under load before we started the data migration process.
+
+### Fate Smiles
+
+I worked all day on Friday to re-deploy CBT and recreate the tests we ran previously.  This time I was able to use all 10 of the drives in each node.  I also bumped up the number of clients to maintain an average of roughly 1 FIO client with an io_depth of 128 per OSD.  The first 3 node test looked good.  With 10 OSDs per node, We were achieving roughly proportional (IE higher) performance relative to the previous tests.  I knew I wasn't going to have much time to do proper scaling tests, so I immediately bumped up from 3 nodes to 10 nodes. I also scaled the PG count at the same time and used CBT to deploy a new cluster.  At 3 nodes I saw 63GiB/s for 4MB random reads.  At 10 Nodes, I saw 213.5GiB/s.  That's almost linear scaling at 98.4%.  It was at this point that I knew that things were finally taking a turn for the better. Of the 68 nodes for this cluster, only 63 were up at that time.  The rest were down for maintenance to fix various issues.  I split the cluster roughly in half, with 32 nodes (320 OSDs) in one half, and 31 client nodes running 10 FIO process each in the other half.  I watched as CBT built the cluster over roughly a 7-8 minute period of time.  The initial write prefill looked really good.  My heart soared.  We were reading data at 635 GiB/s.  We broke 15 million 4k random read IOPS.  While this may not seem impressive compared to the individual NVMe drives, these were the highest numbers I had ever seen for a ~300 OSD Ceph cluster.
```

Comment:
s/recreate/repeat/
io_depth in backticks
I.e.
Suggest a space between 63 GiB/s here and elsewhere